src/try_with_timeout.jl

```diff
@@ -75,15 +75,25 @@ end
 """
 function try_with_timeout(f, timeout, ::Type{T}=Any) where {T}
     ch = Channel{T}(0)
-    x = TimedOut(ch)
-    timer = Timer(tm -> !isready(ch) && close(ch, TimeoutException(timeout)), timeout)
+    timer = Timer(timeout)
     @samethreadpool_spawn begin
+        _ch = $ch
```

Comment:
IMO i would make this shadow the old name, since (I assume) you believe that using the old name in the spawned code is a potential performance hazard. By shadowing it, you make accidentally using the wrong name impossible.
```suggestion
        ch = $ch
```

Reply:
For some reason, this aliasing somehow confuses type inference. Here is code warntype using `ch`:
![image](https://github.com/user-attachments/assets/de39bccf-4daa-4ee4-966f-b738aa9dbd9d)

And here with `_ch`
![image](https://github.com/user-attachments/assets/1868b200-f1b7-4794-a9f3-773ca0b16495)